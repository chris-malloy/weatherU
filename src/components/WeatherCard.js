import React from 'react';
import PropTypes from 'prop-types';
import { CardPanel, Row, Col } from 'react-materialize';

const WeatherCard = (props) => {
  var iconUrl = `https://openweathermap.org/img/w/${props.icon}.png`;
  var jacketMsg;
  var umbrellaMsg;
  if(props.temp > 80) {
    jacketMsg = "Stay home, jacket!";
  } else if (props.temp > 60) {
    jacketMsg = "Okay, maybe a light jacket";
  } else if (props.temp > 40) {
    jacketMsg = "Oh, you gonna need a jacket today";
  } else if (props.temp > 20) {
    jacketMsg = "What, it's winter already???";
  } else if (props.temp > 0) {
    jacketMsg = "STAY INSIDE";
  };
  
  var code = props.code.toString() // to use string proto includes()
  if (code.includes('2')) {
    umbrellaMsg = "Oh no, thunder! Bring 5 umbrellas"
  } else if (code.includes('3')) {
    umbrellaMsg = "Just a drizzle today!";
  } else if (code.includes('5')) {
    umbrellaMsg = "Umbrella time, it's one of those days";
  } else if (code.includes('6')) {
    umbrellaMsg = "Snow Day!!!";
  } else if (code.includes('7')) {
    umbrellaMsg = "Theres' some weird stuff going on today";
  } else if (code.includes('8')) {
    umbrellaMsg = "Happy Day! Leave the umbrella home";
  } else if (code.includes('90')) {
    umbrellaMsg = "Just stay home it's crazy out there";
  };

  return (
      <Col s={12} m={6} l={4}>
        <CardPanel className="grey lighten-3">
          <Row className="center">
            <Col s={6} className="current-temp">
              {props.temp}°
            </Col>
            <Col s={6}>
            <img src={iconUrl} alt="weather icon"/>
            </Col>
            <Col s={12}>
              <p>{jacketMsg}</p>
              <p>{umbrellaMsg}</p>
            </Col>
            <Col s={6} className="date">
            {props.date}
            </Col>
            <Col s={6} className="high-low">
              <Col s={12}>
              {props.high}°
              </Col>
              <Col s={12}>
                {props.low}°
              </Col>
            </Col>
          </Row>
        </CardPanel>
    </Col>
  );
};

WeatherCard.propTypes = {
  temp: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  high: PropTypes.number.isRequired,
  low: PropTypes.number.isRequired,
  code: PropTypes.number.isRequired 
}

export default WeatherCard;