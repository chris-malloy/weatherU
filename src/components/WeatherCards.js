import React from 'react';
import PropTypes from 'prop-types';
import WeatherCard from './WeatherCard';

const WeatherCards = (props) => {
  return props.weatherList.map((data, i) => {
    return <WeatherCard key={i} temp={0} icon='' date='' high={0} low={0} />
  });
};

WeatherCards.propTypes = {
  weatherList: PropTypes.array.isRequired
}

export default WeatherCards;