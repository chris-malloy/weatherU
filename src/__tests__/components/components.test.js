import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WeatherCards from '../../components/WeatherCards';
import WeatherCard from '../../components/WeatherCard'

configure({ adapter: new Adapter() });

describe('Components', () => {
  describe('WeatherCards', () => {
    const props = {
      weatherList: ['weatherList']
    }
    // const weatherCards = mount(<WeatherCards {...props} />);

    // it('renders correctly', () => {
    //   expect(weatherCards).toMatchSnapshot();
    // });

    // it('to have a `WeatherCard` component', () => {
    //   expect(weatherCards.find('WeatherCard').exists()).toBe(true);
    // });
  });

  describe('WeatherCard', () => {
    const props = {
      temp: 0,
      icon: 'icon',
      date: 'date',
      low: 0,
      high: 0
    }
    const weatherCard = shallow(<WeatherCard {...props} />);

    it('renders correctly', () => {
      expect(weatherCard).toMatchSnapshot();
    });
  });
});

// TODO 
// need to fix testing of WeatherCards Component