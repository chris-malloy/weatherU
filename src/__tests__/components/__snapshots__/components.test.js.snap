// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components WeatherCard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    code={0}
    date="date"
    high={0}
    icon="icon"
    low={0}
    temp={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CardPanel
        className="grey lighten-3"
>
        <Row
                className="center"
        >
                <Col
                        className="current-temp"
                        s={6}
                >
                        0
                        °
                </Col>
                <Col
                        s={6}
                >
                        <img
                                alt="weather icon"
                                src="https://openweathermap.org/img/w/icon.png"
                        />
                </Col>
                <Col
                        s={12}
                >
                        <p />
                        <p />
                </Col>
                <Col
                        className="date"
                        s={6}
                >
                        date
                </Col>
                <Col
                        className="high-low"
                        s={6}
                >
                        <Col
                                s={12}
                        >
                                0
                                °
                        </Col>
                        <Col
                                s={12}
                        >
                                0
                                °
                        </Col>
                </Col>
        </Row>
</CardPanel>,
      "l": 4,
      "m": 6,
      "s": 12,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row
          className="center"
>
          <Col
                    className="current-temp"
                    s={6}
          >
                    0
                    °
          </Col>
          <Col
                    s={6}
          >
                    <img
                              alt="weather icon"
                              src="https://openweathermap.org/img/w/icon.png"
                    />
          </Col>
          <Col
                    s={12}
          >
                    <p />
                    <p />
          </Col>
          <Col
                    className="date"
                    s={6}
          >
                    date
          </Col>
          <Col
                    className="high-low"
                    s={6}
          >
                    <Col
                              s={12}
                    >
                              0
                              °
                    </Col>
                    <Col
                              s={12}
                    >
                              0
                              °
                    </Col>
          </Col>
</Row>,
        "className": "grey lighten-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              className="current-temp"
              s={6}
>
              0
              °
</Col>,
            <Col
              s={6}
>
              <img
                            alt="weather icon"
                            src="https://openweathermap.org/img/w/icon.png"
              />
</Col>,
            <Col
              s={12}
>
              <p />
              <p />
</Col>,
            <Col
              className="date"
              s={6}
>
              date
</Col>,
            <Col
              className="high-low"
              s={6}
>
              <Col
                            s={12}
              >
                            0
                            °
              </Col>
              <Col
                            s={12}
              >
                            0
                            °
              </Col>
</Col>,
          ],
          "className": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                0,
                "°",
              ],
              "className": "current-temp",
              "s": 6,
            },
            "ref": null,
            "rendered": Array [
              0,
              "°",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <img
                alt="weather icon"
                src="https://openweathermap.org/img/w/icon.png"
/>,
              "s": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "weather icon",
                "src": "https://openweathermap.org/img/w/icon.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p />,
                <p />,
              ],
              "s": 12,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "date",
              "className": "date",
              "s": 6,
            },
            "ref": null,
            "rendered": "date",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Col
                  s={12}
>
                  0
                  °
</Col>,
                <Col
                  s={12}
>
                  0
                  °
</Col>,
              ],
              "className": "high-low",
              "s": 6,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    0,
                    "°",
                  ],
                  "s": 12,
                },
                "ref": null,
                "rendered": Array [
                  0,
                  "°",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    0,
                    "°",
                  ],
                  "s": 12,
                },
                "ref": null,
                "rendered": Array [
                  0,
                  "°",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CardPanel
          className="grey lighten-3"
>
          <Row
                    className="center"
          >
                    <Col
                              className="current-temp"
                              s={6}
                    >
                              0
                              °
                    </Col>
                    <Col
                              s={6}
                    >
                              <img
                                        alt="weather icon"
                                        src="https://openweathermap.org/img/w/icon.png"
                              />
                    </Col>
                    <Col
                              s={12}
                    >
                              <p />
                              <p />
                    </Col>
                    <Col
                              className="date"
                              s={6}
                    >
                              date
                    </Col>
                    <Col
                              className="high-low"
                              s={6}
                    >
                              <Col
                                        s={12}
                              >
                                        0
                                        °
                              </Col>
                              <Col
                                        s={12}
                              >
                                        0
                                        °
                              </Col>
                    </Col>
          </Row>
</CardPanel>,
        "l": 4,
        "m": 6,
        "s": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Row
            className="center"
>
            <Col
                        className="current-temp"
                        s={6}
            >
                        0
                        °
            </Col>
            <Col
                        s={6}
            >
                        <img
                                    alt="weather icon"
                                    src="https://openweathermap.org/img/w/icon.png"
                        />
            </Col>
            <Col
                        s={12}
            >
                        <p />
                        <p />
            </Col>
            <Col
                        className="date"
                        s={6}
            >
                        date
            </Col>
            <Col
                        className="high-low"
                        s={6}
            >
                        <Col
                                    s={12}
                        >
                                    0
                                    °
                        </Col>
                        <Col
                                    s={12}
                        >
                                    0
                                    °
                        </Col>
            </Col>
</Row>,
          "className": "grey lighten-3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                className="current-temp"
                s={6}
>
                0
                °
</Col>,
              <Col
                s={6}
>
                <img
                                alt="weather icon"
                                src="https://openweathermap.org/img/w/icon.png"
                />
</Col>,
              <Col
                s={12}
>
                <p />
                <p />
</Col>,
              <Col
                className="date"
                s={6}
>
                date
</Col>,
              <Col
                className="high-low"
                s={6}
>
                <Col
                                s={12}
                >
                                0
                                °
                </Col>
                <Col
                                s={12}
                >
                                0
                                °
                </Col>
</Col>,
            ],
            "className": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  0,
                  "°",
                ],
                "className": "current-temp",
                "s": 6,
              },
              "ref": null,
              "rendered": Array [
                0,
                "°",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="weather icon"
                  src="https://openweathermap.org/img/w/icon.png"
/>,
                "s": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "weather icon",
                  "src": "https://openweathermap.org/img/w/icon.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <p />,
                  <p />,
                ],
                "s": 12,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "date",
                "className": "date",
                "s": 6,
              },
              "ref": null,
              "rendered": "date",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Col
                    s={12}
>
                    0
                    °
</Col>,
                  <Col
                    s={12}
>
                    0
                    °
</Col>,
                ],
                "className": "high-low",
                "s": 6,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      0,
                      "°",
                    ],
                    "s": 12,
                  },
                  "ref": null,
                  "rendered": Array [
                    0,
                    "°",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      0,
                      "°",
                    ],
                    "s": 12,
                  },
                  "ref": null,
                  "rendered": Array [
                    0,
                    "°",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
